frontend/datetime.lua

```diff
@@ -0,0 +1,297 @@
+        if twelve_hour_clock then
+            if os.date("%p", seconds) == "AM" then
+                -- @translators This is the time in the morning using a 12-hour clock (%I is the hour, %M the minute).
+                return os.date(_("%I:%M AM"), seconds)
+            else
+                -- @translators This is the time in the afternoon using a 12-hour clock (%I is the hour, %M the minute).
+                return os.date(_("%I:%M PM"), seconds)
+            end
+        else
+            -- @translators This is the time using a 24-hour clock (%H is the hour, %M the minute).
+            return os.date(_("%H:%M"), seconds)
+        end
+    end
+else
+    function datetime.secondsToHour(seconds, twelve_hour_clock, pad_with_spaces)
+        if twelve_hour_clock then
+            if os.date("%p", seconds) == "AM" then
+                if pad_with_spaces then
+                    -- @translators This is the time in the morning using a 12-hour clock (%_I is the hour, %M the minute).
+                    return os.date(_("%_I:%M AM"), seconds)
+                else
+                    -- @translators This is the time in the morning using a 12-hour clock (%-I is the hour, %M the minute).
+                    return os.date(_("%-I:%M AM"), seconds)
+                end
+            else
+                if pad_with_spaces then
+                    -- @translators This is the time in the afternoon using a 12-hour clock (%_I is the hour, %M the minute).
+                    return os.date(_("%_I:%M PM"), seconds)
+                else
+                    -- @translators This is the time in the afternoon using a 12-hour clock (%-I is the hour, %M the minute).
+                    return os.date(_("%-I:%M PM"), seconds)
+                end
+            end
+        else
+            if pad_with_spaces then
+                -- @translators This is the time using a 24-hour clock (%_H is the hour, %M the minute).
+                return os.date(_("%_H:%M"), seconds)
+            else
+                -- @translators This is the time using a 24-hour clock (%-H is the hour, %M the minute).
+                return os.date(_("%-H:%M"), seconds)
+            end
+        end
+    end
+end
+
+--- Converts timestamp to a date string
+---- @int seconds number of seconds
+---- @use_locale if true allows to translate the date-time string, if false return "%Y-%m-%d time"
+---- @treturn string date string
+function datetime.secondsToDate(seconds, use_locale)
+    seconds = seconds or os.time()
+    if use_locale then
+        local wday  = os.date("%a", seconds)
+        local month = os.date("%b", seconds)
+        local day   = os.date("%d", seconds)
+        local year  = os.date("%Y", seconds)
+
+        -- This format string has to be adjusted by translators
+        -- for german the format string would be "%1 %3 %2 %4"
+        return T(C_("%1 name of day, %2 name of month, %3 day, %4 year", "%1 %2 %3 %4"),
```

Comment:
What do translators have to do here?

Reply:
Sort it :)
I the US I think it should be "Mo Apr/23/2022", in German I know it is "Mo 23. Apr. 2022" and so on.
Therefore the question should be, what should we use as the default string in the line above?

Reply:
"%1 %2/%3/%4" ?

Reply:
Oh goodness, you're using context in a way no one's ever used it in millions of years. :laughing:

Reply:
The main problem is your confusing comment, so there's a chance it might be comprehensible to translators.

Reply:
Anyway, context is supposed to be something like `Date string` with the comment explaining `%1 name of day, %2 name of month, %3 day, %4 year` and then it works the way everyone expects it to work.